src/functions.jl

```diff
@@ -22,3 +22,17 @@ invlog1(b::Real, x::Real) = b ≥ zero(b) && x ≥ zero(x) ? b^x : throw(DomainE
 invlog1(b, x) = b^x
 
 invlog2(b, x) = x^inv(b)
+
+
+function invdivrem((q, r), divisor)
+    res = muladd(q, divisor, r)
+    abs(r) ≤ abs(divisor) && _issamesign(r, res) || throw(DomainError((q, r), "inverse for divrem(x) is not defined at this point"))
+    return res
+end
+
+function invfldmod((q, r), divisor)
+    abs(r) ≤ abs(divisor) && _issamesign(r, divisor) || throw(DomainError((q, r), "inverse for fldmod(x) is not defined at this point"))
+    return muladd(q, divisor, r)
+end
+
+_issamesign(x, y) = iszero(x) || iszero(y) || sign(x) == sign(y)
```

Comment:
Maybe it would be a bit more correct to call this function
```suggestion
_is_product_nonnegative(x, y) = iszero(x) || iszero(y) || sign(x) == sign(y)
```
or something shorter? In the first two cases the sign is not necessarily the same.

Reply:
No, same sign does not include zero:
```julia
julia> mod(0.0, -10.0)
-0.0

julia> mod(0.0, 10.0)
0.0
```
It's just that
```julia
julia> 0.0 === -0.0
false

julia> 0 === -0
true
```